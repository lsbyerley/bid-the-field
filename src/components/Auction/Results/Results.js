import { getAuctionResults, getPlayerFromBid } from '../../../lib/auctionUtils';

const Results = ({ bids = [], players = [] }) => {
  const auctionResults = getAuctionResults(bids);

  return (
    <div className='grid grid-cols-1 gap-6 md:grid-cols-3'>
      {Object.keys(auctionResults).map((owner) => {
        const ownerResults = auctionResults[owner];
        const totalBidAmount = ownerResults.reduce(
          (total, bid) => total + bid.amount,
          0
        );
        return (
          <div
            key={owner}
            className='rounded-lg md:row-span-3 card card-compact bg-base-200'
          >
            <div className='card-body'>
              <div>
                <div className='flex justify-between pb-2 mb-4 text-lg font-medium border-b'>
                  <span className='mr-2 text-lg truncate text-info'>
                    {owner}
                  </span>
                  <span className='text-success'>
                    ${Number.parseFloat(totalBidAmount).toFixed(2)}
                  </span>
                </div>
                <dl className='mt-2 divide-y'>
                  {ownerResults.map((bid) => (
                    <div
                      key={bid.id}
                      className='flex justify-between py-3 text-sm font-medium'
                    >
                      <dt className=''>
                        {getPlayerFromBid(players, bid.player_id)}
                      </dt>
                      <dd className=''>
                        ${Number.parseFloat(bid.amount).toFixed(2)}
                      </dd>
                    </div>
                  ))}
                  {!ownerResults ||
                    (!ownerResults?.length && (
                      <div className='px-2 py-3'>
                        <dt className=''>no winning bids</dt>
                      </div>
                    ))}
                </dl>
              </div>
            </div>
          </div>
        );
      })}
      {!Object.keys(auctionResults)?.length && (
        <div className='shadow-lg alert'>
          <div>
            <svg
              xmlns='http://www.w3.org/2000/svg'
              className='flex-shrink-0 w-6 h-6 stroke-current'
              fill='none'
              viewBox='0 0 24 24'
            >
              <path
                strokeLinecap='round'
                strokeLinejoin='round'
                strokeWidth='2'
                d='M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z'
              />
            </svg>
            <span>No auction results! code: 89</span>
          </div>
        </div>
      )}
    </div>
  );
};

export default Results;
