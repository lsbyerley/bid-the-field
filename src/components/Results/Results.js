import { getAuctionResults, getPlayerFromBid } from '@/lib/auctionUtils';

const Results = ({ bids = [], players = [] }) => {
  const auctionResults = getAuctionResults(bids);

  return (
    <>
      <div className='grid grid-cols-1 gap-6 mb-8 md:grid-cols-3'>
        {Object.keys(auctionResults).map((owner) => {
          const ownerResults = auctionResults[owner];
          const totalBidAmount = ownerResults.winningBids.reduce(
            (total, bid) => total + bid.amount,
            0
          );
          return (
            <div
              key={owner}
              className='rounded-lg md:row-span-3 card card-compact bg-base-100'
            >
              <div className='card-body'>
                <div>
                  <div className='flex justify-between pb-2 mb-4 text-lg font-medium border-b'>
                    <span className='mr-2 text-lg truncate text-info'>
                      {ownerResults?.profile?.name ||
                        ownerResults?.profile?.email}
                    </span>
                    <span className='text-success'>
                      ${Number.parseFloat(totalBidAmount).toFixed(2)}
                    </span>
                  </div>
                  <dl className='mt-2 overflow-y-scroll divide-y max-h-96'>
                    {ownerResults.winningBids.map((bid) => {
                      const player = getPlayerFromBid(players, bid.player_id);
                      return (
                        <div
                          key={bid.id}
                          className='flex justify-between py-3 text-sm font-medium'
                        >
                          <dt className=''>
                            {player?.full_name ||
                              player?.short_name ||
                              `na:${bid.player_id}`}
                          </dt>
                          <dd className=''>
                            ${Number.parseFloat(bid.amount).toFixed(2)}
                          </dd>
                        </div>
                      );
                    })}
                    {!ownerResults ||
                      (!ownerResults?.winningBids?.length && (
                        <div className='px-2 py-3'>
                          <dt className='text-warning'>no winning bids</dt>
                        </div>
                      ))}
                  </dl>
                </div>
              </div>
            </div>
          );
        })}
      </div>
      {!Object.keys(auctionResults)?.length && (
        <div className='justify-center rounded-lg alert bg-base-100'>
          <div>
            <svg
              xmlns='http://www.w3.org/2000/svg'
              className='flex-shrink-0 w-6 h-6 stroke-current'
              fill='none'
              viewBox='0 0 24 24'
            >
              <path
                strokeLinecap='round'
                strokeLinejoin='round'
                strokeWidth='2'
                d='M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z'
              />
            </svg>
            <span>No rosters or results yet!</span>
          </div>
        </div>
      )}
    </>
  );
};

export default Results;
