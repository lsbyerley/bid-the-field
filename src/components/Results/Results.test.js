import React from 'react';
import { render, screen } from 'test-utils';
import Results from './Results';

jest.mock('@/lib/auctionUtils', () => ({
  ...jest.requireActual('@/lib/auctionUtils'),
  getAuctionResults: jest.fn((props) => {
    return {
      ownerOne: [
        {
          id: 'test',
          auction_id: 'test',
          player_id: '1',
          amount: 25,
          owner: 'ownerOne',
        },
      ],
      ownerTwo: [
        {
          id: 'test',
          auction_id: 'test',
          player_id: '2',
          amount: 15,
          owner: 'ownerTwo',
        },
      ],
    };
  }),
}));

const defaultProps = (overrides) => {
  return {
    bids: [
      {
        id: 'test',
        owner: 'ownerOne',
        amount: 25,
        player_id: '1',
        auction_id: 'test',
      },
      {
        id: 'test',
        owner: 'ownerTwo',
        amount: 20,
        player_id: '1',
        auction_id: 'test',
      },
    ],
    players: [
      {
        id: '1',
        full_name: 'test-full-1',
        short_name: 'test-short-1',
      },
      {
        id: '2',
        full_name: 'test-full-2',
        short_name: 'test-short-2',
      },
    ],
    ...overrides,
  };
};

const renderResults = (props) => {
  const utils = render(<Results {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('Results', () => {
  it('should render and match snapshot', () => {
    const { home } = renderResults(defaultProps());
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
