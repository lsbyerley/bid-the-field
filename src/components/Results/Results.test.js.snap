// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render and match snapshot 1`] = `
<div
  class="grid grid-cols-1 gap-6 md:grid-cols-3"
>
  <div
    class="rounded-lg md:row-span-3 card card-compact bg-base-100"
  >
    <div
      class="card-body"
    >
      <div>
        <div
          class="flex justify-between pb-2 mb-4 text-lg font-medium border-b"
        >
          <span
            class="mr-2 text-lg truncate text-info"
          >
            ownerOne
          </span>
          <span
            class="text-success"
          >
            $
            25.00
          </span>
        </div>
        <dl
          class="mt-2 divide-y"
        >
          <div
            class="flex justify-between py-3 text-sm font-medium"
          >
            <dt
              class=""
            >
              test-full-1
            </dt>
            <dd
              class=""
            >
              $
              25.00
            </dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
  <div
    class="rounded-lg md:row-span-3 card card-compact bg-base-100"
  >
    <div
      class="card-body"
    >
      <div>
        <div
          class="flex justify-between pb-2 mb-4 text-lg font-medium border-b"
        >
          <span
            class="mr-2 text-lg truncate text-info"
          >
            ownerTwo
          </span>
          <span
            class="text-success"
          >
            $
            15.00
          </span>
        </div>
        <dl
          class="mt-2 divide-y"
        >
          <div
            class="flex justify-between py-3 text-sm font-medium"
          >
            <dt
              class=""
            >
              test-full-2
            </dt>
            <dd
              class=""
            >
              $
              15.00
            </dd>
          </div>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
